editor/src/components/canvas/dom-lookup.ts

```diff
@@ -50,39 +55,76 @@ export function findFirstParentWithValidUID(
 }
 
 export function getValidTargetAtPoint(
-  validTemplatePaths: Array<string>,
+  componentMetadata: JSXMetadata,
+  selectedViews: Array<TemplatePath>,
+  hiddenInstances: Array<TemplatePath>,
+  validTemplatePathsForLookup: Array<string> | 'no-filter',
   point: WindowPoint | null,
+  canvasScale: number,
+  canvasOffset: CanvasVector,
 ): TemplatePath | null {
   if (point == null) {
     return null
   }
-  const elementsUnderPoint = document.elementsFromPoint(point.x, point.y)
-  for (const element of elementsUnderPoint) {
-    const foundValidtemplatePath = findFirstParentWithValidUID(validTemplatePaths, element)
-    if (foundValidtemplatePath != null) {
-      return TP.fromString(foundValidtemplatePath)
-    }
-  }
-  return null
+  return (
+    getAllTargetsAtPoint(
+      componentMetadata,
+      selectedViews,
+      hiddenInstances,
+      validTemplatePathsForLookup,
+      point,
+      canvasScale,
+      canvasOffset,
+    )[0] ?? null
+  )
 }
 
-export function getAllTargetsAtPoint(point: WindowPoint | null): Array<TemplatePath> {
+export function getAllTargetsAtPoint(
+  componentMetadata: JSXMetadata,
+  selectedViews: Array<TemplatePath>,
+  hiddenInstances: Array<TemplatePath>,
+  validTemplatePathsForLookup: Array<string> | 'no-filter',
+  point: WindowPoint | null,
+  canvasScale: number,
+  canvasOffset: CanvasVector,
+): Array<TemplatePath> {
   if (point == null) {
     return []
   }
+  const pointOnCanvas = windowToCanvasCoordinates(canvasScale, canvasOffset, point)
+  const getElementsUnderPointFromAABB = Canvas.getAllTargetsAtPoint(
+    componentMetadata,
+    selectedViews,
+    hiddenInstances,
+    pointOnCanvas.canvasPositionRaw,
+    [TargetSearchType.All],
+    false,
+    'loose',
+  )
   const elementsUnderPoint = document.elementsFromPoint(point.x, point.y)
-  return stripNulls(
+  const elementsFromDOM = stripNulls(
     elementsUnderPoint.map((element) => {
-      const validTPs = findParentSceneValidPaths(element)
-      if (validTPs != null) {
-        const foundValidtemplatePath = findFirstParentWithValidUID(validTPs, element)
-        if (foundValidtemplatePath != null) {
-          return TP.fromString(foundValidtemplatePath)
-        }
+      const foundValidtemplatePath = findFirstParentWithValidUID(
+        validTemplatePathsForLookup,
+        element,
+      )
+      if (foundValidtemplatePath != null) {
+        return TP.fromString(foundValidtemplatePath)
+      } else {
+        return null
       }
-      return null
     }),
   )
+
+  return getElementsUnderPointFromAABB
+    .filter((foundElement) => {
+      if (!foundElement.canBeFilteredOut) {
+        return true
```

Comment:
I'm finding this very hard to read because of the combination of `filter` removing elements that return `false` combined with the `!foundElement.canBeFilteredOut` returning `true` inside the `if` statement :confused: